cmd/api/api.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"sda-pipeline/internal/broker"
+	"sda-pipeline/internal/config"
+	"sda-pipeline/internal/database"
+
+	"github.com/gorilla/mux"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var Conf *config.Config
+var err error
+
+func main() {
+	Conf, err = config.NewConfig("api")
+	if err != nil {
+		log.Fatal(err)
+	}
+	Conf.API.MQ, err = broker.NewMQ(Conf.Broker)
+	if err != nil {
+		log.Fatal(err)
+	}
+	Conf.API.DB, err = database.NewDB(Conf.Database)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	go func() {
+		connError := Conf.API.MQ.ConnectionWatcher()
+		log.Error(connError)
+		os.Exit(1)
```

Comment:
It feels sort of scary to just exit here. Doesn't that mean that any mq connection error will kill the application regardless of the main thread state? Or can't this trigger once the connection is established?

Reply:
It's what we currently use in all the other as well, but I agree we should probably replace this with a reconnect function.